dashboard/app/helpers/aichat_sagemaker_helper.rb

```diff
@@ -13,17 +13,20 @@ def self.create_sagemaker_client
     Aws::SageMakerRuntime::Client.new
   end
 
-  def self.get_instructions(system_prompt, retrieval_contexts)
+  def self.get_instructions(system_prompt, level_system_prompt, retrieval_contexts)
     instructions = ""
-    instructions = system_prompt + " " unless system_prompt.empty?
+    instructions = level_system_prompt + " " unless level_system_prompt.empty?
+    instructions << (system_prompt + " ") unless system_prompt.empty?
     instructions << retrieval_contexts.join(" ") if retrieval_contexts
     instructions
   end
 
-  def self.format_inputs_for_sagemaker_request(aichat_model_customizations, stored_messages, new_message)
+  def self.format_inputs_for_sagemaker_request(aichat_model_customizations, stored_messages, new_message, level_id)
     selected_model_id = aichat_model_customizations[:selectedModelId]
     # Add system prompt and retrieval contexts if available to inputs as part of instructions that will be sent to model.
-    instructions = get_instructions(aichat_model_customizations[:systemPrompt], aichat_model_customizations[:retrievalContexts])
+    # Get level system prompt that will be prepended to student system prompt.
+    level_system_prompt = Level.find(level_id).properties["aichat_settings"]["levelSystemPrompt"]
```

Comment:
Minor: might be safer to use `dig`, i.e. `.properties.dig('aichat_settings', 'levelSystemPrompt')` in the rare case that `aichat_settings` doesn't exist for this level? Also should we check and/or rescue an error if the level_id isn't for a valid level? Actually I think if we do something like `Level.find_by(id: level_id)&.properties.dig(...)` then we should be safe (i.e. won't throw if there is no level, will just return `nil`).

Reply:
Thanks! Love the `dig` method!
Got a lint error `dashboard/app/helpers/aichat_sagemaker_helper.rb:28:59: W: [Correctable] Lint/SafeNavigationChain: Do not chain ordinary method call after safe navigation operator` for your last suggestion so broke it up into:
```
level_properties = Level.find(level_id)&.properties
level_system_prompt = level_properties.dig('aichat_settings', 'levelSystemPrompt') unless  level_properties.nil?
```

Reply:
Ah interesting - I wonder if that means we need to do `properties&.dig()`? In any case, breaking it up works fine too. Just a callout though that using `Level.find_by` instead of `Level.find` may be safer as it seems like the former returns `nil` if the model is not found while the latter throws an error.

Reply:
Yes, the `properties&.dig` works - thanks!
However, using `find_by` resulted in an error
```
ArgumentError at /aichat/chat_completion
========================================

Unsupported argument type: 17503 (Integer)

> To access an interactive console with this error, point your browser to: /__better_errors

app/helpers/aichat_sagemaker_helper.rb, line 21
-----------------------------------------------

``` ruby
   20       # Get level system prompt that will be prepended to student system prompt.
>  21       level_system_prompt = Level.find_by(level_id)&.properties&.dig('aichat_settings', 'levelSystemPrompt')
   22       instructions = get_instructions(aichat_model_customizations[:systemPrompt], level_system_prompt, 
```

Reply:
Just added that if there is no `level_system_prompt` defined in `aichat_settings`, set to an empty string. Also added a unit test.

Reply:
Yeah `find_by` requires a column name to do the look up by, so you'll need to do something like `Level.find_by(id: level_id)`

Reply:
Learning a lot about ruby/rails in this PR. Thanks!